.github/workflows/test.yaml

```diff
@@ -0,0 +1,78 @@
+name: test
+
+on:
+  pull_request:
```

Comment:
Should we set this to be only runnable by maintainers somehow? I am wondering if the `go generate ./...` step will use a lot of our runner minutes.

Reply:
If it's fast enough <5min, it's okay if it's ok even on all commits

Reply:
There's additional controls at the repo level where it will require maintainer approval before running. I'm not sure if that's enabled but it's another knob to turn